experiment/simulation/js/main.js

```diff
@@ -1 +1,293 @@
+	const ctx = canvas.getContext("2d");
+
+	const fill = "black";
+	const lineWidth = 1.5;
+	const fps = 20;
+
+	
+	const plateStartX = 250;
+	const plateWidth = 400;
+	const upperPlateStartY = 340;
+	const bottomPlateStartY = 700;
+	const plateHeight = 120;
+	const rodWidth = 240;
+	const upperRodHeight = 300;
+	const lowerRodHeight = 360;
+	const boxWidth = 320;
+	const boxStarty = 500;
+	const breakLength = 80;
+	const gap = 10;
+	const benchHeight = 100;
+
+	const boxColor = "#2C9AD1"; // virtual labs color
+	const plateColor = "#176696"; // 
+	const rodColor = "grey";
+	const benchColor = "#643D01";
+
+	
+	setAll();
+	drawStatic();
+	graph();
+
+	function drawStatic()
+	{
+		ctx.clearRect(0, 0, canvas.width, canvas.height);
+		ctx.fillStyle = fill;
+		ctx.lineWidth = lineWidth;
+		ctx.lineCap = "round";
+		ctx.lineJoin = "round";
+		ctx.font = "50px Arial";
+
+		ctx.fillText("UTM Machine",20,100);
+		ctx.fillText("Load",620,upperPlate[1][1]-110);
+		ctx.lineWidth = 5;
+		ctx.beginPath();
+		ctx.moveTo(600,upperPlate[1][1]-150);
+		ctx.lineTo(600,upperPlate[1][1]-50);
+		ctx.lineTo(580,upperPlate[1][1]-70);
+		ctx.moveTo(600,upperPlate[1][1]-50);
+		ctx.lineTo(620,upperPlate[1][1]-70);
+		ctx.stroke();
+		drawObject(ctx,upperPlate, plateColor);
+		drawObject(ctx,bottomPlate, plateColor);
+		drawObject(ctx,upperRod, rodColor);
+		drawObject(ctx,lowerRod, rodColor);
+		drawObject(ctx,bench, benchColor);
+	}
+	function draw()
+	{
+		if(upperPlate[0][1] + plateHeight < boxStarty)
+		{
+			drawStatic();
+			drawObject(ctx,box,boxColor);
+			move(upperRod , 1);
+			move(upperPlate,1);
+			tmHandle = 	window.setTimeout(draw, 1000 / fps);
+		}	
+		else
+		{
+			updateChart();
+			
+		}	
+		
+	}
+
+	function graph()
+	{
+		coordinates.push({
+			x : 0,
+			y : 0
+		})
+		chart = new CanvasJS.Chart("chartContainer", 
+		{
+			title :{
+				text: "Stress v/s Strain" 
+			},
+			axisY: {
+				includeZero: true,
+				title: "Stress (MPa)"
+			},
+			axisX: {
+				includeZero: true,
+				title: "%Strain"
+			},      
+			data: [{
+				type: "line",
+				dataPoints: coordinates
+			}]
+		});
+	}
+
+
+	
+	function updateChart()
+	{
+		
+		let xVal = strain[step];            
+		let yVal = stress[step];        
+		coordinates.push({
+					x: xVal,
+					y: yVal
+			});
+		if (coordinates.length > 100)
+			coordinates.shift();
+		document.getElementById("stress").innerHTML = stress[step].toString() + " Mpa";
+		document.getElementById("strain").innerHTML = strain[step].toString();
+		document.getElementById("load").innerHTML = load[step].toString() + " N";
+		document.getElementById("deformation").innerHTML = deformation[step].toString() + " mm";
+		if(step<load.length)
+		{
+			chart.render();
+			step++;
+		}
+		if(step == load.length)
+		{
+			drawStatic();
+			drawObject(ctx,brokenBoxPart1, boxColor);
+			drawObject(ctx,brokenBoxPart2, boxColor);
+			window.clearTimeout(tmHandle);
+		}
+		else
+		{
+			tmHandle = 	window.setTimeout(draw, 5000 / fps);
```

Comment:
This code will execute when step<load.length
Is this the intended outcome?

Reply:
yes sir it is the intended outcome.